students/jstrauss123/lesson03/mailroom.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+# function - send thank you
+def fun_sendty():
+    full_name = input("Please provide full name of donor: ")
+    while full_name == "list":
+        print(donor_list)
+        full_name = input("Please provide full name of donor: ")
+    if full_name in donor_list:
+        #print("full name entered is: ", full_name)
+        idx_num = donor_list.index(full_name)
+        #print("index is: ", idx_num)
+        #str1 = "donamt_" + str(idx_num)
+        #print("str1 value is: ", str1)
+        #print(" list value is equal to: ", str1[-1])
+        # if donor in list send thank you email
+        print("Dear {}, Thank you so much for your generous contribution".format(donor_list[idx_num]))
+    # if user not in donor_list add new donor and prompt for contribution amount
+    else: 
+        donor_list.append(full_name)
+        idx_num = donor_list.index(full_name)
+        new_donamt = input("Enter donation amount for new donor: ")
+        newamt = float(new_donamt)
+        print("Dear {}, Thank you so much for your generous contribution".format(donor_list[idx_num]))
+        # code here to create new contribution list for new donor
+        #donamt_<insert index here>.append(newamt)
+    #print("response is: ", response)    
+    #if response_ty == "list"
+    #    print(donor_list)
+    #if response_ty in donor_list:
+
+# function - create report
+def fun_report():
+    #arr6 = [['Joey', 30, 199.99], ['Jan', 36, 29999.97], ['Bob', 55, 999999.99]] 
+    donor_len = len(donor_list)
+    count1 = 0
+    print("")
+    print("{:15} |{:>15} |{:>2} |{:>15}".format("Donor Name","Total Given","Num Gifts","Average Gifts" ))
+    while count1 < donor_len:
+        
+        print("{:15} ${:>15} {:>2} ${:>15}".format(donor_list[count1], 25000, 2, 10000))
+        count1 += 1
+    print("")
+        
+
+# main
+if __name__ == "__main__":
+    #donor_list = [["Mickey Mouse", 100, 150, 100], ["Minnie Mouse", 50, 50], ["Ron Jones", 100],["Donald Duck", "25"], ["Busy Bear", "10", "10", "10"]]
+    donor_list = ["Mickey Mouse", "Minnie Mouse", "Ron Jones", "Donald Duck", "Busy Beaver"]
+    donamt_0 = ["100", "100", "100"]
+    donamt_1 = ["50", "50", "100"]
+    donamt_2 = ["100", "100", "100"]
+    donamt_3 = ["10", "10", "10"]
+    donamt_4 = ["5", "5"]
```

Comment:
I'm not sure why this is setup this way? 

Please review guidelines:
https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom_tutorial.html

Highly recommend going with the data structure in there.

Reply:
I am fully confused by what appears to be three separate python cert URL's. I never saw this tutorial before. I was directed to use this URL:
https://startlearning.uw.edu/courses/course-v1:UW+PYTHON210+2018_Winter/courseware

some times this URL redirects to one of the following:
https://uwpce-pythoncert.github.io/PythonCertDevel/modules  which the title indicates it's Pythoncert 5.0 and other times it links to:
http://uwpce-pythoncert.github.io/IntroPython2016a which the title indicates it's PythonCert 1.3


In any event, my mailroom.py was replaced by mailroom1.py after I figured out how to use a list in a list on my own. Please let me know if I should replace the existing mailroom.py with mailroom1.py.

Reply:
Yes please resubmit with a single file updated so that we know what to review.

https://uwpce-pythoncert.github.io/PythonCertDevel/modules/ is supplemental material for the course. It contains everything you see on the startlearning.uw.edu plus extensive materials like the guidelines I linked for you.

Reply:
Thanks will resubmit with the original names.